### PR TITLE
Feat/file handler

### PR DESCRIPTION
**Pull request by BlackRece**
**Describe your changes**
Completed refactoring File Handling to unify and simplify the process of loading and saving json files to and from editors.
Have also updated the projectile editor to use the new File Handler class to save and load projectile pattern json files.

**Screenshots**
Code to use File Handler
image

Open File Dialog Window
image

Save File Dialog Window
image

**Additional context**
Add any other context about your changes here.

### CODE REVIEW

File: MH-Engine/Components/ProjectileEditor.cpp [hunk 1]
```diff
@@ -61,36 +57,57 @@ void ProjectileEditor::SpawnEditorWindow(const Graphics& gfx, ConstantBuffer<Mat
...
+		->ShowDialog()		// Show the dialog.
+		->StoreDialogResult();	// Store the result.
+
+	// Check if the file object has a file path/name.
+	if (foLoad->HasPath()) 
+		JsonLoading::LoadJson(m_vecManagers, foLoad->GetFullPath());
 	else
 		m_sSelectedFile = "Open File Failed";
+
+	// Smile. :D
```
Comment (juliusz-kaczmarek) [inline MH-Engine/Components/ProjectileEditor.cpp:80]: :)

File: MH-Engine/Components/ProjectileEditor.cpp [hunk 2]
```diff
@@ -61,36 +57,57 @@ void ProjectileEditor::SpawnEditorWindow(const Graphics& gfx, ConstantBuffer<Mat
...
+		->UseSaveDialog()	// Choose the dialog to use.
+		->ShowDialog()		// Show the dialog.
+		->StoreDialogResult();	// Store the result.
+	
+	// Check if the file object has a file path/name.
+	if (foSave->HasPath())
+		// Save the file.
+		JsonLoading::SaveJson(m_vecManagers, foSave->GetJsonPath());
+	
+	// Smile. :D
```
Comment (juliusz-kaczmarek) [inline MH-Engine/Components/ProjectileEditor.cpp:110]: :)

**Review by juliusz-kaczmarek — APPROVED** [review]
Looks fine to me, remove smile comments if they are unnecessary. 😄🔫😈

**Review by kyle-robinson — APPROVED** [review]
Seems to be working fine. Had to make a few changes since there were issues with UI loading but otherwise, all is in order.

Just make sure to use the ```ErrorLogger``` in the future to properly log errors with HRESULT's 😎 👍🏻

**Pull request merged**